tests/eas/acceptance.py

```diff
@@ -85,6 +88,52 @@ def log_result(data, log_fh):
     logging.info(result_str)
     log_fh.write(result_str)
 
+@wrapt.decorator
+def experiment_test(wrapped_test, instance, args, kwargs):
+    for experiment in instance.executor.experiments:
+        tasks = experiment['wload'].tasks.keys()
+        try:
+            wrapped_test(experiment, tasks, *args, **kwargs)
+        except AssertionError as e:
+            trace_relpath = os.path.join(experiment['out_dir'], "trace.dat")
+            logging.error("Check trace file: " + os.path.abspath(trace_relpath))
+
+class EasTest(LisaTest):
+    """
+    Base class for EAS tests
+    """
+
+    set_is_big_little = True
+
+    @classmethod
+    def setUpClass(cls, *args, **kwargs):
+        conf_file = os.path.join(os.path.dirname(os.path.abspath(__file__)),
+                                 cls.conf_basename)
+
+        super(EasTest, cls)._init(conf_file, *args, **kwargs)
+
+    @memoized
+    def get_multi_assert(self, experiment, task_filter=""):
+        tasks = experiment['wload'].tasks.keys()
+        return SchedMultiAssert(experiment['out_dir'],
+                                self.te.topology,
+                                [t for t in tasks if task_filter in t])
+
+    def get_start_time(self, experiment):
+        start_times_dict = self.get_multi_assert(experiment).getStartTime()
+        return min([t['starttime'] for t in start_times_dict.values()])
+
+    @experiment_test
+    def test_first_cpu(self, experiment, tasks):
```

Comment:
Maybe this is overkilling for a base class. I know that in the original implementation KP was checking this condition in each test but, given the opportunity of this refactoring, i'm wondering if it really makes sense to check this condition multiple times.

Moreover, since this is a base class, should not we consider the further addition of new EasTests where instead for example we expect the task to start on a little CPU (e.g. negative boosted tasks)?

Perhaps I would prefer to have a little bit of replicated code in subclasses instead of having mandatory tests enforced in all them.

Maybe a convenient midway solution is to turn this into a local private method, e.g. _test_first_cpu, which can than be simply called by a test_first_cpu defined by a derived class?